docs/_tutorials/pipeline.md

```diff
@@ -269,17 +269,17 @@ by DeepSpeed:
 * `partition_method="uniform"` balances the number of layers per stage.
 
 ### Memory-Efficient Model Construction
-Building a `Sequential` and providing it `PipelineModule` is a convenient way
+Building a `Sequential` container and providing it to a `PipelineModule` is a convenient way
 of specifying a pipeline parallel model. However, this approach encounters
-scalability issues for massive models. Starting from a `Sequential` allocates
-the model in CPU memory redundantly by every worker. A machine with 16 GPUs
+scalability issues for massive models. In this approach each worker replicates
+the whole model in CPU memory. For example, a machine with 16 GPUs
 must have as much local CPU memory as 16 times the model size.
 
 DeepSpeed provides a `LayerSpec` class that delays the construction of
 modules until the model layers have been partitioned across workers. Then,
-the modules are built on the GPU that owns the layer.
+each GPU allocates only the modules assigned to it.
```

Comment:
Is this supposed to communicate that when `LayerSpec` is used, not only that each GPU allocates only the layers it's assigned to or are you also trying to say that no CPU allocation happens at all?

Or should I have rewritten it to say:

> Then each worker will allocate only the layers it's assigned to. So continuing the example from the previous paragraph, a machine with 16 GPUs will need to allocate a total 1x model size on its CPU.

I think the two paras have GPU and CPU mixed up.

Reply:
That's correct, `LayerSpec` does not allocate modules until the partitioning is complete. At that point, each rank allocates only the layers assigned to it and they are then moved to the GPU. This lets us build distributed models that wouldn't fit in CPU memory.

Reply:
So is it better to leave the original modification of this PR, or replace with the more exemplifying:

> Then each worker will allocate only the layers it's assigned to. So continuing the example from the previous paragraph, a machine with 16 GPUs will need to allocate a total of 1x model size on its CPU, compared to 16x in the LayerSpec example.

Reply:
OK, I went ahead with the longer version. Please double check that it's still correct.